skyportal/handlers/api/internal/ephemeris.py

```diff
@@ -17,15 +18,42 @@ def get(self, telescope_id):
         else:
             time = ap_time.Time.now()
 
-        try:
-            telescope_id = int(telescope_id)
-        except TypeError:
-            return self.error(f'Invalid telescope id: {telescope_id}, must be integer.')
+        ephemerides = None
+
+        if telescope_id is not None:
+            try:
+                telescope_id = int(telescope_id)
+            except ValueError as e:
+                return self.error(f'Invalid value for Telescope id: {e.args[0]}')
+            telescope = Telescope.query.filter(Telescope.id == telescope_id).first()
+            if telescope is None:
+                return self.error('No Telescope with this id')
+            else:
+                if telescope.fixed_location is not True:
+                    return self.error('Telescope is not fixed')
+                else:
+                    ephemerides = telescope.ephemeris(time)
+        else:
+            telescope_ids = self.get_query_argument('telescopeIds', None)
+            if telescope_ids is not None:
+                try:
+                    telescope_ids = [int(t) for t in telescope_ids.split(',')]
+                except ValueError as e:
+                    return self.error(f'Invalid telescopeIds format: {e.args[0]}')
+
+                telescopes = Telescope.query.filter(
+                    Telescope.id.in_(telescope_ids)
+                ).all()
+                if len(telescopes) > 16:
+                    telescopes = telescopes[:16]
+            else:
+                telescopes = Telescope.query.all()
+                if len(telescopes) > 16:
```

Comment:
I suggest setting a DEFAULT_MAX_TELESCOPES_TO_RETURN or similar at the top of the file and make it changible with a command line argument. But otherwise do something similar to this.

Reply:
Ok i've set a variable on top of the file, made some other changes and fixes some frontend issues. I'm not sure how to handle the part of allowing to change that max value from the command line. Is this done somewhere in SkyPortal ?